src/bin/inx-chronicle/cli.rs

```diff
@@ -255,55 +256,37 @@ impl ClArgs {
-                                        db.update_all_analytics(index, prev_analytics).await?;
-                                        influx_db
-                                            .insert_all_analytics(timestamp, index, prev_analytics.clone())
-                                            .await?;
-                                    } else {
-                                        let analytics = db.get_all_analytics(index).await?;
-                                        prev_analytics = Some(analytics.clone());
-                                        influx_db.insert_all_analytics(timestamp, index, analytics).await?;
-                                    }
 
-                                    #[cfg(feature = "metrics")]
-                                    {
-                                        let elapsed = start_time.elapsed();
-                                        influx_db
-                                            .insert(chronicle::db::collections::metrics::AnalyticsMetrics {
-                                                time: chrono::Utc::now(),
-                                                milestone_index: index,
-                                                analytics_time: elapsed.as_millis() as u64,
-                                                chronicle_version: std::env!("CARGO_PKG_VERSION").to_string(),
-                                            })
-                                            .await?;
-                                    }
-                                    tracing::info!("Finished analytics for milestone {}", index);
-                                } else {
-                                    tracing::info!("No milestone in database for index {}", index);
-                                }
+                        let mut handle = shutdown_signal.subscribe();
+                        join_set.spawn(async move {
+                            tokio::select! {
+                                res = fill_analytics(db, influx_db, start_milestone, end_milestone) => res?,
+                                _ = handle.recv() => {},
                             }
                             Ok(())
                         });
                         // Account for inclusive end
                         start_milestone = end_milestone + 1;
                     }
-                    while let Some(res) = join_set.join_next().await {
-                        // Panic: Acceptable risk
-                        res.unwrap()?;
+
+                    // We wait for either the interrupt signal or the completion of filling the analytics.
+                    loop {
+                        tokio::select! {
+                            _ = crate::process::interrupt_or_terminate() => {
+                                tracing::info!("received ctrl-c or terminate");
+                                shutdown_signal.send(())?;
```

Comment:
I might be a little slow today, but who is receiving this signal?

Reply:
each created task in the `JoinSet`

Reply:
Right, thanks!